typing/gprinttyp.ml

```diff
@@ -521,8 +540,48 @@ module Digraph = struct
   let labelf fmt = labelk Fun.id fmt
   let labelr fmt = labelk Decoration.make fmt
 
-  let add_node explicit_d color id tynode dg =
-    let d = labelf "<SUB>%a</SUB>" Pp.prettier_index id in
+  (* Use unicode superscript digit to circumvent graphviz limited support for
+     superscript. *)
+  let superscript_digit ppf n =
+    let s = match n with
+    | 1 -> "¹"
+    | 2 -> "²"
+    | 3 -> "³"
+    | 0 -> "⁰"
+    | 4 -> "⁴"
+    | 5 -> "⁵"
+    | 6 -> "⁶"
+    | 7 -> "⁷"
+    | 8 -> "⁸"
+    | _ -> "⁹"
+    in
+    Format.pp_print_string ppf s
+
+  let rec superscript ppf n =
+    if n < 10 then
+      superscript_digit ppf n
+    else begin
+      superscript ppf (n/10);
+      superscript_digit ppf (n mod 10)
+    end
+
+  let superscript_level ppf lvl =
+    (* avoid a dependency on Btype *)
+    if lvl = Ident.highest_scope then Format.pp_print_string ppf "ʷ"
```

Comment:
space then [combining infinity](https://www.fileformat.info/info/unicode/char/1ab2/index.htm) would also work.

Reply:
This is better indeed (at least while we still have only one generic level), thanks for the suggestion!